internal/driver/virtualdriver.go

```
@@ -198,3 +198,13 @@ func deleteVirtualResources(driver *VirtualDriver, deviceName string) error {
 		return nil
 	}
 }
+
+func (d *VirtualDriver) Discover() error {
+	d.lc.Warn("Driver's Discover function didn't implement")
+	return nil
```

please return an error and remove the Warn, because SDK will log this error

---

Updated to return an error.